packages/solid/src/solidrouter.ts

```diff
@@ -82,13 +82,14 @@ function withSentryRouterRoot(Root: Component<RouteSectionProps>): Component<Rou
       const rootSpan = getActiveRootSpan();
 
       if (rootSpan) {
-        const { op, description } = spanToJSON(rootSpan);
+        const { op, description, data } = spanToJSON(rootSpan);
 
         // We only need to update navigation spans that have been created by
         // a browser back-button navigation (stored as `-1` by solid router)
         // everything else was already instrumented correctly in `useBeforeLeave`
         if (op === 'navigation' && description === '-1') {
           rootSpan.updateName(name);
+          rootSpan.setAttribute(SEMANTIC_ATTRIBUTE_SENTRY_SOURCE, data && data[SEMANTIC_ATTRIBUTE_SENTRY_SOURCE]);
```

Comment:
I'm fairly sure the source here is always "url" but for now I just set it to what it was before updating the name. I'll let @andreiborza confirm so that we can save some bytest here.